lib/saml/settings_service.rb

```diff
@@ -28,10 +28,14 @@ def saml_settings
       end
 
       def merged_saml_settings
-        OneLogin::RubySaml::IdpMetadataParser.new.parse(metadata, settings: settings)
-      rescue => e
-        log_message_to_sentry("SAML::SettingService failed to parse SAML metadata: #{e.message}", :error)
```

Comment:
When `metadata` was `nil`, this used to just return the `settings` that we passed in.  In `ruby-saml 1.5.0`, an `ArgumentError` is raised if `metadata` does not contain an `IDPSSODescriptor ` xml element ([Source](https://github.com/onelogin/ruby-saml/blob/e9d454b5bf1d203c53fd008a7c2200aff9e4f7d6/lib/onelogin/ruby-saml/idp_metadata_parser.rb#L106)).

I made this change to tests pass but I'm actually debating if raising might actually be better...

Reply:
I would think raising makes sense, this would happen if ID.me is down for some reason right?

Reply:
yes.  in the past the metadata endpoint has been offline and we got a 404 `text/html` response with an ID.me 404 page.